hoodie-client/src/main/java/com/uber/hoodie/io/HoodieCreateHandle.java

```diff
@@ -69,8 +70,16 @@ public HoodieCreateHandle(HoodieWriteConfig config, String commitTime, HoodieTab
       HoodiePartitionMetadata partitionMetadata = new HoodiePartitionMetadata(fs, commitTime,
           new Path(config.getBasePath()), FSUtils.getPartitionPath(config.getBasePath(), partitionPath));
       partitionMetadata.trySave(TaskContext.getPartitionId());
-      this.storageWriter = HoodieStorageWriterFactory
-          .getStorageWriter(commitTime, getStorageWriterPath(), hoodieTable, config, writerSchema);
+
+      String type = config.getFileType();
+      if (!type.equalsIgnoreCase("orc")) {
+        this.storageWriter = HoodieStorageWriterFactory
+            .getStorageWriter(commitTime, getStorageWriterPath(), hoodieTable, config, writerSchema);
+      } else {
+        TypeDescription orcSchema = TypeDescription.fromString(config.getOrcStringSchema());
+        this.storageWriter = HoodieStorageWriterFactory
```

Comment:
since the config already contains the OrcStringSchema, can we handle this difference internally within `HoodieStorageWriterFactory.getStorageWriter()`.. Ideally, the handle classes should just call `getStorageWriter` not a orc specific method like `getOrcStorageWriter`